src/sync/impl/partial_sync_helper.hpp

```diff
@@ -0,0 +1,71 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2017 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#ifndef partial_sync_helper_hpp
+#define partial_sync_helper_hpp
+
+#include <functional>
+#include <string>
+#include <unordered_map>
+
+namespace realm {
+    
+class Realm;
+class List;
+
+using PartialSyncResultCallback = void(List results, std::exception_ptr error);
```

Comment:
Recommend using `std::error_code` to pass errors by value rather than `std::exception_ptr`.

Reply:
It looks like this is used in conjunction with our existing object notification callbacks, which already use `std::execption_ptr` for this purpose. Trying to change it here would require being able to translate an `std::exception_ptr` to a `std::error_code`, which seems impractical.